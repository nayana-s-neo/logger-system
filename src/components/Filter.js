import React, { useEffect, useState } from 'react'
import {useNavigate,useLocation} from "react-router-dom";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

function Filter({applicationTypes, actionTypes, resetSearch, getNewDate}) {
  
    const [fromDate, setFromDate] = useState();
    const [toDate, setToDate] = useState();
    let navigate = useNavigate();
    let currentPath  = useLocation();
    let path = currentPath.search;
    const query = new URLSearchParams(path);
    useEffect(() => {
      if(query.get('fromDate'))
          setFromDate(new Date(query.get('fromDate')));
      if(query.get('toDate'))
        setToDate(new Date(query.get('toDate')));  
    },[path]);


    const handleNavigation = (event) => {
        const formData = new FormData(event.currentTarget);
        event.preventDefault();   
        let urlStr = "?"; 
        for (let [key, value] of formData.entries()) {
          if(value)
          {
            if(urlStr.length === 1 && urlStr === '?')
              urlStr = urlStr +key+"="+value;
            else
              urlStr = urlStr + '&'+key+"="+value;
          } 
        }
        if(fromDate && toDate)
        {
            if(urlStr.length === 1 && urlStr === '?')
              urlStr = urlStr+"fromDate="+getNewDate(fromDate)+"&toDate="+getNewDate(toDate);
            else
              urlStr = urlStr+"&fromDate="+getNewDate(fromDate)+"&toDate="+getNewDate(toDate);
        }
        navigate("/logger"+urlStr);
      }
  return (
    <form data-testid="search-logger-form" onSubmit={handleNavigation}> 
          <div className="searchContainer">
            <div className="m-2">
              <label>Log ID</label>
              <input defaultValue={query.get('logId')} type="text" data-testid="log-id" name="logId" className="form-control" onChange={resetSearch}/>
            </div>
            <div className="m-2">
              <label>Application Type</label>
              <select name="applicationType" defaultValue={query.get('applicationType')} data-testid="application-type"  className="form-control" onChange={resetSearch}>
                <option value="">Select Application Type</option>
                {
                  applicationTypes.map((item) => {
                    return(
                      <option key={item} value={item}>{item}</option>
                    )
                  })
                }
              </select>
            </div>
            <div className="m-2">
              <label>Application Id</label>
              <input type="text" defaultValue={query.get('applicationId')} data-testid="application-id"  name="applicationId" className="form-control" onChange={resetSearch}/>
            </div>
            <div className="m-2">
              <label>Action Type</label>
              <select name="actionType" defaultValue={query.get('actionType')} data-testid="action-type"  className="form-control" onChange={resetSearch}>
              <option value="">Select Action Type</option>
                {
                  actionTypes.map((item) => {
                    return(
                      <option key={item} value={item}>{item}</option>
                    )
                  })
                }
              </select>
            </div>
            <div className="m-2">
              <label>From Date</label>
              <DatePicker className="form-control" selected={fromDate} onChange={(date:Date) => {setFromDate(date); resetSearch()}} className="form-control"/>
            </div>
            <div className="m-2">
              <label>To Date</label>
              <DatePicker  className="form-control" selected={toDate} onChange={(date:Date) => {setToDate(date); resetSearch()} } className="form-control" />
            </div>
            
            <button type="submit" data-testid="search-logger" className="btn btn-info">Search Logger</button>
            
          </div>
          </form>

  )
}

export default Filter