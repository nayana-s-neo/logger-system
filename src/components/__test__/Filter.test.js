import { render } from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import Filter from '../Filter';

const applicationTypes = ['LEASE_REGISTRATION','ADD_COMPANY_EMPLOYEE','CERT_PROP_OWNERSHIP'];
const actionTypes = ['SUBMIT_APPLICATION','ADD_EMPLOYEE','INITIATE_APPLICATION'];

const resetSearch = jest.fn();
const getNewDate = jest.fn();

describe('filter section',() => {
  beforeEach(() => {
    jest.resetAllMocks();
 });


it('renders filter section', () => {
  const {container} = render(<Router><Filter applicationTypes={applicationTypes} actionTypes={actionTypes}/></Router>);
  expect(container).toMatchSnapshot();
});

it('check filter elements present', async () => {
    const {getByTestId} = render(<Router><Filter applicationTypes={applicationTypes} actionTypes={actionTypes} resetSearch={resetSearch} getNewDate={getNewDate} /></Router>);
    expect(getByTestId('search-logger')).toBeInTheDocument();
    expect(getByTestId('log-id')).toBeInTheDocument();
    expect(getByTestId('application-id')).toBeInTheDocument();
    expect(getByTestId('action-type')).toBeInTheDocument();
    expect(getByTestId('application-type')).toBeInTheDocument();
    expect(getByTestId('search-logger')).toBeInTheDocument();
});
});