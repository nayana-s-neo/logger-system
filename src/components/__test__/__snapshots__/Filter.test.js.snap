// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter section renders filter section 1`] = `
<div>
  <form
    data-testid="search-logger-form"
  >
    <div
      class="searchContainer"
    >
      <div
        class="m-2"
      >
        <label>
          Log ID
        </label>
        <input
          class="form-control"
          data-testid="log-id"
          name="logId"
          type="text"
          value=""
        />
      </div>
      <div
        class="m-2"
      >
        <label>
          Application Type
        </label>
        <select
          class="form-control"
          data-testid="application-type"
          name="applicationType"
        >
          <option
            value=""
          >
            Select Application Type
          </option>
          <option
            value="LEASE_REGISTRATION"
          >
            LEASE_REGISTRATION
          </option>
          <option
            value="ADD_COMPANY_EMPLOYEE"
          >
            ADD_COMPANY_EMPLOYEE
          </option>
          <option
            value="CERT_PROP_OWNERSHIP"
          >
            CERT_PROP_OWNERSHIP
          </option>
        </select>
      </div>
      <div
        class="m-2"
      >
        <label>
          Application Id
        </label>
        <input
          class="form-control"
          data-testid="application-id"
          name="applicationId"
          type="text"
          value=""
        />
      </div>
      <div
        class="m-2"
      >
        <label>
          Action Type
        </label>
        <select
          class="form-control"
          data-testid="action-type"
          name="actionType"
        >
          <option
            value=""
          >
            Select Action Type
          </option>
          <option
            value="SUBMIT_APPLICATION"
          >
            SUBMIT_APPLICATION
          </option>
          <option
            value="ADD_EMPLOYEE"
          >
            ADD_EMPLOYEE
          </option>
          <option
            value="INITIATE_APPLICATION"
          >
            INITIATE_APPLICATION
          </option>
        </select>
      </div>
      <div
        class="m-2"
      >
        <label>
          From Date
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="m-2"
      >
        <label>
          To Date
        </label>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="form-control"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="btn btn-info"
        data-testid="search-logger"
        type="submit"
      >
        Search Logger
      </button>
    </div>
  </form>
</div>
`;
